packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -401,6 +396,12 @@ export default function(
     const context = needsContext
       ? getMaskedContext(workInProgress, unmaskedContext)
       : emptyObject;
+
+    // Instantiate twice to help detect side-effects.
+    if (debugRenderPhaseSideEffects) {
+      new ctor(props, context); // eslint-disable-line no-new
+    }
```

Comment:
So this should call the constructor twice, right?

Reply:
Yes. It is called the second time right below:
```js
const instance = new ctor(props, context);
```